File: 372.超级次方.js
/*
 * @lc app=leetcode.cn id=372 lang=javascript
 *
 * [372] 超级次方
 */

// @lc code=start
/**
 * @param {number} a
 * @param {number[]} b
 * @return {number}
 */
var superPow = function (a, b) {
  // 1140 是欧拉数
  let numB = b[0];
  for (let index = 1; index < b.length; index++) {
    const element = b[index];
    numB = (numB * 10 + element) % 1140;
  }
  numB = (numB % 1140) + 1140;
  let numA = 1;
  for (let time = 0; time < numB; time++) {
    numA *= a;
    numA %= 1337;
  }
  return numA;
};
// @lc code=end

File: 372.超级次方.rs
/*
 * @lc app=leetcode.cn id=372 lang=rust
 *
 * [372] 超级次方
 */

// @lc code=start
impl Solution {
    pub fn super_pow(a: i32, b: Vec<i32>) -> i32 {
        let b = b.into_iter().fold(0, |s, bi| (s * 10 + bi) % 1140);
        (0..b).fold(1, |s, _| s * a as i64 % 1337) as i32
    }
}
// @lc code=end

